pkg/storage/cacher_test.go

```diff
@@ -165,267 +124,135 @@ func TestList(t *testing.T) {
 		case "baz":
 			expected = podBaz
 		default:
-			t.Errorf("unexpected item: %v", item)
+			t.Errorf("Unexpected item: %v", item)
 		}
 		if e, a := *expected, item; !reflect.DeepEqual(e, a) {
-			t.Errorf("expected: %#v, got: %#v", e, a)
+			t.Errorf("Expected: %#v, got: %#v", e, a)
 		}
 	}
+}
 
-	close(fakeClient.WatchResponse)
+func verifyWatchEvent(t *testing.T, w watch.Interface, eventType watch.EventType, eventObject runtime.Object) {
+	select {
+	case event := <-w.ResultChan():
+		if e, a := eventType, event.Type; e != a {
+			t.Errorf("Expected: %s, got: %s", eventType, event.Type)
+		}
+		if e, a := eventObject, event.Object; !api.Semantic.DeepDerivative(e, a) {
+			t.Errorf("Expected (%s): %#v, got: %#v", eventType, e, a)
+		}
+	case <-time.After(util.ForeverTestTimeout):
```

Comment:
Do we need this?  I typically like having the tests get stuck in the exact location where it is to trace through the stack.